faf-commons-data/src/test/java/com/faforever/commons/replay/LoadReplayLoaderTest.java

```diff
@@ -0,0 +1,118 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+class LoadReplayLoaderTest {
+
+  @TempDir
+  public Path temporaryFolder;
+
+  private final ObjectMapper objectMapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+  @Test
+  public void parseBinary01() throws CompressorException, IOException {
+    assertDoesNotThrow(
+      () -> {
+        Path fafReplayFile = temporaryFolder.resolve("TestCommands01.fafreplay");
+        Files.copy(Objects.requireNonNull(getClass().getResourceAsStream("/replay/TestCommands01.fafreplay")), fafReplayFile);
+        ReplayContainer fafReplayContainer = ReplayLoader.loadFAFReplayFromDisk(fafReplayFile);
+      }
+    );
+  }
+
+  @Test
+  public void parseBinary02() throws CompressorException, IOException {
+    assertDoesNotThrow(
+      () -> {
+        Path fafReplayFile = temporaryFolder.resolve("TestModeratorEvents.fafreplay");
+        Files.copy(Objects.requireNonNull(getClass().getResourceAsStream("/replay/TestModeratorEvents.fafreplay")), fafReplayFile);
+        ReplayContainer fafReplayContainer = ReplayLoader.loadFAFReplayFromDisk(fafReplayFile);
+      }
+    );
+  }
+
+  @Test
+  public void parseBinary03() throws CompressorException, IOException {
+    assertDoesNotThrow(
+      () -> {
+        Path fafReplayFile = temporaryFolder.resolve("zstd_reference.fafreplay");
+        Files.copy(Objects.requireNonNull(getClass().getResourceAsStream("/replay/zstd_reference.fafreplay")), fafReplayFile);
+        ReplayContainer fafReplayContainer = ReplayLoader.loadFAFReplayFromDisk(fafReplayFile);
+      }
+    );
+  }
+
+  @Test
+  public void parseBinary04() throws CompressorException, IOException {
+    assertDoesNotThrow(
+      () -> {
+        Path fafReplayFile = temporaryFolder.resolve("test.fafreplay");
+        Files.copy(Objects.requireNonNull(getClass().getResourceAsStream("/replay/test.fafreplay")), fafReplayFile);
+        ReplayContainer fafReplayContainer = ReplayLoader.loadFAFReplayFromDisk(fafReplayFile);
+      }
+    );
+  }
+
+  @Test
+  public void compareBinary01() throws CompressorException, IOException {
+    Path fafReplayFile = temporaryFolder.resolve("22338092.fafreplay");
+    Files.copy(Objects.requireNonNull(getClass().getResourceAsStream("/replay/load/22338092.fafreplay")), fafReplayFile);
+
+    Path scfaReplayFile = temporaryFolder.resolve("22338092.scfareplay");
+    Files.copy(Objects.requireNonNull(getClass().getResourceAsStream("/replay/load/22338092.scfareplay")), scfaReplayFile);
+
+    ReplayContainer fafReplayContainer = ReplayLoader.loadFAFReplayFromDisk(fafReplayFile);
+    ReplayContainer scfaReplayContainer = ReplayLoader.loadSCFAReplayFromDisk(scfaReplayFile);
+
+    assertEquals(scfaReplayContainer.body().events().size(), fafReplayContainer.body().events().size());
+    assertArrayEquals( scfaReplayContainer.bytes(), fafReplayContainer.bytes());
```

Comment:
I would rather not expose the bytes of the replay. But how would I verify this property?